### ISSUE TITLE
accumulate multiple iterations of the same task for parallel execution

### ISSUE THREAD

**Issue by asalt** [ploomber/ploomber#607]
I am dynamically creating tasks based on available files. The only changes to each task are the input file and expected output. I'd like to run these dynamically generated tasks in parallel. How can I aggregate these tasks together ahead of time to form one big DAG?These are the "pseudo-methods" I have in mind to achieve this based on the docs.

```
all_dags = list()
for f in files:
    env = make_env(f)
    dag = DAGSpec('pipeline.yaml', env=env).to_dag() # pipeline.yaml will have default params that are applied to all tasks,
                                                     # the newly created `env` contains the input file and expected output(?)
    all_dags.append(dag)
    # dag.build() # this blocks accumulation of additional tasks

# make one big dag?
combine_dags(all_dags).build()  # allow parallel execution
```

or append to one dag? 
```
dag = None
for f in files:
    env = make_env(f)
    if dag is None:
        dag = DAGSpec('pipeline.yaml', env=env).to_dag()
    dag["task"].append(dag)  # add to list of tasks to run

dag.build() # parallel execution
```

**Comment by edublancas** [comment]
Hi, thanks for your feedback.

I ran a quick experiment to see how we could support this, something like this:

```python
from ploomber import DAG
from ploomber.spec import DAGSpec

dag = DAG()

for f in files:
   # pass existing dag so tasks are appended to it
    DAGSpec('pipeline.yaml', env=dict(file=f).to_dag(dag=dag)
```

However, this simple approach did not work since DAG expects task names to be unique, we can still support it but we'd have to make some other changes (like allowing DAGSpec to modify the names in the spec).

My recommendation, for now, is to use the [Python API directly](https://github.com/ploomber/projects/tree/master/python-api-examples/examples). The Python API is well suited for these kinds of cases where you want to create DAGs dynamically, let me know if this works for you.

**Comment by asalt** [comment]
Thanks for the response.  I've have success by "appending" jobs to a single DAG. This works very easily.

```
from ploomber import DAG
from ploomber.executors import Parallel
from ploomber.tasks import ShellScript

script_file = pathlib.Path("some-shell-script.sh")

dag = DAG(executor=Parallel())

for f in files:
    file_specific_env = get_file_specific_env(f) 
    product = get_file_specific_product(f)
    ShellScript(script_file, 
                dag = dag,
                product = product,
                name = get_file_specific_name(f)
    )

dag.build() # runs all of the jobs that are not complete, and in parallel
```

**Comment by edublancas** [comment]
glad to know you got it working! don't hesitate to open another issue if you have more questions!
